kernel/src/thread/mod.rs

```diff
@@ -107,7 +108,12 @@ impl Thread {
     }
 }
 
-/// Allocates a new tid for the new thread
-pub fn allocate_tid() -> Tid {
-    TID_ALLOCATOR.fetch_add(1, Ordering::SeqCst)
+/// Allocates a new tid for the new posix thread
+pub fn allocate_posix_tid() -> Tid {
+    POSIX_TID_ALLOCATOR.fetch_add(1, Ordering::SeqCst)
+}
+
+/// Allocates a new tid for the new kernel thread
+pub fn allocate_kernel_tid() -> Tid {
+    KERNEL_TID_ALLOCATOR.fetch_add(1, Ordering::SeqCst)
```

Comment:
Are there any user of kernel TIDs? I am not convinced that kernel TIDs are useful in the first place. Holding a `Arc<Thread>` or `Weak<Thread>` is almost as space efficient as holding a `Tid`.

Reply:
Kernel TIDs are never used except in some debug code. I have moved `Tid` from `Thread` to `PosixThread`, so kernel thread won't have Tids now.